### PR TITLE
Better cgroup handling in metrics monitoring

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: src/main/java/com/metamx/metrics/cgroups/CgroupDiscoverer.java [hunk 1]
```diff
@@ -20,5 +20,6 @@
 
 public interface CgroupDiscoverer
 {
-  Path discover(String cgroup, long pid);
+  // Returns a path for a specific cgroup
```
Comment (leventov) [inline src/main/java/com/metamx/metrics/cgroups/CgroupDiscoverer.java]: Make it javadoc please

File: src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.metamx.metrics.PidDiscoverer;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ProcPidCgroupDiscoverer implements CgroupDiscoverer
+{
+  private final ProcCgroupDiscoverer delegate;
+
+  //    final File pidCgroups = new File(new File(proc, Long.toString(pid)), "cgroup")
```
Comment (leventov) [inline src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java]: Please remove

File: src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.metamx.metrics.cgroups;
```
Comment (leventov) [inline src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java]: Please add copyright header

Comment (drcrallen) [reply src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java]: arg, good catch.

File: src/main/java/com/metamx/metrics/cgroups/CgroupDiscoverer.java (hunk 1 above)
Comment (drcrallen) [reply src/main/java/com/metamx/metrics/cgroups/CgroupDiscoverer.java]: Done

File: src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java (hunk 2 above)
Comment (drcrallen) [reply src/main/java/com/metamx/metrics/cgroups/ProcPidCgroupDiscoverer.java]: Fixed, thanks!

**Pull request merged**